### PR TITLE
1st refactor

### PR DESCRIPTION
**Pull request by Zinoex**
- Add sanity checks to system construction
- Remove unnecessary dependencies (there are still a couple more I want to remove, but that'll come when I separate the examples and library better).
- Add result structs. This way all algorithms can return eta, beta, and the barrier (and allow a function to calculate psafe), while for many PWC algorithms, we also return the region local beta values. Later I want to add timing information to these structures such that this is returned too. Note that while it appears some of the structs are replicated, this is a conscious choice - they will change for different reasons depending on the algorithm and with them separate, you can add/remove data to one result struct without affecting the result structs for other algorithms.

### CODE REVIEW

File: StochasticBarrierFunctions/src/barrier_algorithms/base.jl [hunk 1]
```diff
@@ -0,0 +1,8 @@
+### Algorithm types
```
Comment (rayomaz) [inline StochasticBarrierFunctions/src/barrier_algorithms/base.jl:1]: This is a lot cleaner, thanks. I will update the toolbox description accordingly.

File: StochasticBarrierFunctions/src/barrier_algorithms/beta.jl [hunk 2]
```diff
@@ -4,32 +4,28 @@
...
 
     Threads.@threads for jj in eachindex(regions)
         Xⱼ, Bⱼ = regions[jj], B[jj]
 
-        model = get!(task_local_storage(), "verify_beta_model") do
-            # Using Mosek as the LP solver
-            model = Model(Mosek.Optimizer)
-            set_silent(model)
+        # Using Mosek as the LP solver
+        model = Model(Mosek.Optimizer)
```
Comment (rayomaz) [inline StochasticBarrierFunctions/src/barrier_algorithms/beta.jl:17]: Good call. I remember this caused issues with the GP stuff.

**Review by rayomaz — APPROVED** [review]
Looks good.

File: StochasticBarrierFunctions/src/barrier_algorithms/base.jl (hunk 1 above)
Comment (Zinoex) [reply StochasticBarrierFunctions/src/barrier_algorithms/base.jl:1]: - [ ] Setup website for documentation

Comment (Zinoex) [reply StochasticBarrierFunctions/src/barrier_algorithms/base.jl:1]: What do you mean? I only moved the algorithm struct definitions into their separate files.

Comment (rayomaz) [reply StochasticBarrierFunctions/src/barrier_algorithms/base.jl:1]: I am thinking about making the toolbox paper story more clear

**Pull request merged**